pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -203,15 +202,13 @@ func (r *ReconcileZookeeperCluster) reconcileStatefulSet(instance *zookeeperv1be
 				For details see:
 				https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically
 			*/
-			r.skipSTSReconcile++
-			if r.skipSTSReconcile < 6 {
-				r.log.Info("Waiting for Config Map update to sync...Skipping STS Reconcile")
-				return nil
+			for i := 0; i < 4; i++ {
+				time.Sleep(30 * time.Second)
```

Comment:
Do we really need to sleep 4x for 30 seconds and print the same log message 4x? It would make sense to do it in the case we are checking whether the config map has been updated.

Reply:
We come to the for loop (at ln #205) only after the config map has been updated with the new value and that check is done by this piece of code a few lines earlier:
"
if !r.isConfigMapInSync(instance) {
"
The for loop is to wait for the volume to be updated as there is no way to check this directly by opening the config map volume on each pod in the STS. 
The reason we wait for 2 minutes is based on this:
https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically

Reply:
Hmm.  We should avoid long blocking operations in the reconcile loop, since they may disrupt the operator from from handling changes when running multiple ZK clusters with one operator.

Perhaps we can make a go-client call to the config-map in the container's liveness check rather than relying on teh file-system mount?  This would require adding an addition client to the container image...

Alternatively, if we must really wait 2 minutes, we could store a new field in the ZookeeperCluster Status block, and allow the periodic reconcile loop to replace the sleeps.

Reply:
Will check on getting these options to work....

Reply:
As suggested, tried installing `kubectl` on the zk container and using `kubectl describe <configmap>` to get the value of `CLUSTER_SIZE` variable inside the pod, instead of reading from volume. But it did not work because kubectl command failed in the absence of the right service account, roles and role-bindings being present for accessing the resource.